### PR TITLE
[MINDEXER-164] Fix IOOBEx on GAV remote repositories

### PR DESCRIPTION
**Pull request by fawind**
This is a follow-up on https://github.com/apache/maven-indexer/pull/198 and [MINDEXER-144](https://issues.apache.org/jira/browse/MINDEXER-144), which only fixed the IndexOutOfBound issues for some file types.

The exception can still happen if the local repository contains `_remote.repository` or similar files. Apparently, with Maven 3.0.x, maven creates these file to record where the file was resolved from ([source](https://stackoverflow.com/a/16870552/3388748)).

<details>
  <summary>Full stacktrace for reference:</summary>

```
Caused by: java.lang.IndexOutOfBoundsException: start 56, end 64, length 57
	at java.base/java.lang.AbstractStringBuilder.checkRange(AbstractStringBuilder.java:1794)
	at java.base/java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:675)
	at java.base/java.lang.StringBuilder.append(StringBuilder.java:217)
	at org.apache.maven.index.artifact.M2GavCalculator.getSnapshotGav(M2GavCalculator.java:188)
	at org.apache.maven.index.artifact.M2GavCalculator.pathToGav(M2GavCalculator.java:104)
	at org.apache.maven.index.DefaultArtifactContextProducer.getGavFromPath(DefaultArtifactContextProducer.java:160)
	at org.apache.maven.index.DefaultArtifactContextProducer.getArtifactContext(DefaultArtifactContextProducer.java:83)
	at org.apache.maven.index.DefaultScanner.processFile(DefaultScanner.java:109)
	at org.apache.maven.index.DefaultScanner.scanDirectory(DefaultScanner.java:99)
	at org.apache.maven.index.DefaultScanner.scanDirectory(DefaultScanner.java:91)
	at org.apache.maven.index.DefaultScanner.scanDirectory(DefaultScanner.java:91)
	at org.apache.maven.index.DefaultScanner.scanDirectory(DefaultScanner.java:91)
	at org.apache.maven.index.DefaultScanner.scanDirectory(DefaultScanner.java:91)
	at org.apache.maven.index.DefaultScanner.scanDirectory(DefaultScanner.java:91)
	at org.apache.maven.index.DefaultScanner.scan(DefaultScanner.java:60)
	at org.apache.maven.index.DefaultNexusIndexer.scan(DefaultNexusIndexer.java:291)
	... 8 more
```

</details>

---

https://issues.apache.org/jira/browse/MINDEXER-164

### CODE REVIEW

File: indexer-core/src/test/java/org/apache/maven/index/artifact/M2GavCalculatorTest.java [hunk 1]
```diff
@@ -585,6 +585,8 @@ public void testGav()
         assertNull( gav );
         gav = gavCalculator.pathToGav( "/dev/mbien/hintmod/maven-metadata-local.xml" );
         assertNull( gav );
+        gav = gavCalculator.pathToGav( "/dev/mbien/hintmod/1.0-SNAPSHOT/_remote.repositories" ); // causes MINDEXER-144
+        assertNull( gav );
```
Comment (fawind) [inline indexer-core/src/test/java/org/apache/maven/index/artifact/M2GavCalculatorTest.java:589]: This case failed with `IndexOutOfBoundsException` before the patch.

**Review by cstamas — APPROVED** [review]

**Comment by cstamas** [comment]
master is already prepped for 7.0.0 release (java11, etc), will create 6.2.x branch soon and sort out this.

**Comment by fawind** [comment]
Thanks for taking a look, I updated the PR linking to a separate issue: https://issues.apache.org/jira/browse/MINDEXER-164

**Comment by cstamas** [comment]
Pushed branch https://github.com/apache/maven-indexer/tree/maven-indexer-6.2.x (for upcoming 6.2.2)
@fawind I will handle this, will merge this PR (to master) and will cherry-pick to this branch above as well

**Comment by jira-importer** [comment]
Resolve #518

**Comment by jira-importer** [comment]
Resolve #623

**Comment by jira-importer** [comment]
Resolve #518

**Pull request merged**